File: exercism/haskell/difference-of-squares/Squares.hs
module Squares (sumOfSquares, squareOfSums, difference) where

sumOfSquares :: Integral a => a -> a
sumOfSquares n = sum $ map square [1..n]

squareOfSums :: Integral a => a -> a
squareOfSums n = square $ sum [1..n]

difference :: Integral a => a -> a
difference n = squareOfSums n - sumOfSquares n

square :: Integral a => a -> a
square n = n * n

File: exercism/csharp/difference-of-squares/Squares.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Squares
{
    int N;

    public Squares(int n)
    {
        if (n < 0) throw new ArgumentException();
        N = n;
    }

    public int SumOfSquares()
    {
        return Enumerable.Range(1, N).Select(Square).Sum();
    }

    public int SquareOfSums()
    {
        return Square(Enumerable.Range(1, N).Sum());
    }

    public int DifferenceOfSquares()
    {
        return SquareOfSums() - SumOfSquares();
    }

    private static int Square(int n)
    {
        return n * n;
    }
}
